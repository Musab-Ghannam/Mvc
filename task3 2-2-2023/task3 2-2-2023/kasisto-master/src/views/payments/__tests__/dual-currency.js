/* eslint-env jest */

import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import {
  reduxForm,
  reducer as formReducer
} from 'redux-form'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import DualCurrency from '../dual-currency'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

describe('DualCurrency', () => {
  const createForm = reduxForm({
    form: 'dummy'
  })

  it('renders fiat and money', () => {
    const store = mockStore({
      form: formReducer,
      settings: {}
    })

    const Form = createForm(({ handleSubmit }) => (
      <form onSubmit={handleSubmit}>
        <DualCurrency />
      </form>
    ))
    const component = renderer.create(
      <Provider store={store}>
        <Form />
      </Provider>
    )

    let tree = component.toJSON()

    // expect(Payment.on).toBeCalledWith('ready')

    expect(tree).toMatchSnapshot()
  })
})
