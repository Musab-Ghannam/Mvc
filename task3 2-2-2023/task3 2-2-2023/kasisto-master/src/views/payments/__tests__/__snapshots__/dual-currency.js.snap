// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DualCurrency renders fiat and money 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="c-dual-currency "
  >
    <label
      className="c-currency c-currency--input c-currency--primary u-margin-bottom-none"
    >
      <input
        className="c-currency--input__native_input"
        id="requestedAmount"
        min="0"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="00"
        step="0.01"
        type="number"
        value="NaN"
      />
      <span
        className="c-currency__tail"
      >
        .00
      </span>
      <small
        className="c-currency__currency"
      />
    </label>
    <small
      className="u-margin-bottom"
    >
      <span
        className="icon icon--loading"
      >
        <svg>
          <use
            xlinkHref="#loading"
          />
        </svg>
      </span>
       
       / XMR 
      <br />
      <a
        target="_blank"
      />
    </small>
    <span
      className="c-currency c-currency--secondary"
    >
      <span
        className="c-currency__value"
      >
        0.000000000000
      </span>
      Â 
      <span
        className="c-currency__tail"
      >
        
      </span>
      <span
        className="c-currency__currency"
      >
        XMR
      </span>
    </span>
  </div>
</form>
`;
